app/models/event_stream.rb

```diff
@@ -54,20 +56,21 @@ def self.event_groups
   end
 
   def self.group_and_level(event_type)
-    group, v = event_groups.find { |_k, v| v[:critical].include?(event_type) || v[:detail].include?(event_type) }
-    if group.nil?
-      group, level = :other, :detail
-    else
-      level = v[:detail].include?(event_type) ? :detail : :critical
+    level = :detail # the level is detail as default
+    group = event_groups.find do |_k, value|
+      lvl_found = GROUP_LEVELS.detect { |lvl| value[lvl]&.include?(event_type) }
+      level = lvl_found if lvl_found # update the level variable if found the current level.
+      lvl_found # the level found or nil
     end
+
+    group = group.kind_of?(Array) ? group.first : :other
     return group, level
   end
 
   def self.group_name(group)
-    return nil if group.nil?
-    group = event_groups[group.to_sym]
-    return nil if group.nil?
-    group[:name]
+    return if group.nil?
+    group = event_groups[group]
+    group.nil? ? 'Other' : group[:name]
```

Comment:
This method used to support passed in group as a string.

Reply:
Can we also add test cases here to ensure the behavior is not changed accidentally?

Reply:
> This method used to support passed in group as a string and it returns nil if group is nil.
> Is it intentional to change the behavior?

@lfu Good point,  it is a symbol, but I will add the `group.sym` again because the group column have a type as string( [Line 31 ](https://github.com/ManageIQ/manageiq/pull/17611/files/45bbdd33b2740eccbde5165b394be3e78bbfa082#diff-c68bc7715d55b6fadc383efb49df7164R31)) and when persisted in the virtual column it is converted as string. Someone can do a ``` ```

Reply:
> Can we also add test cases here to ensure the behavior is not changed accidentally?

I think the tests already cover this case.